listeners/FilterTimeslot.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Igniter\Local\Listeners;
+
+use Admin\Models\Orders_model;
+use Carbon\Carbon;
+use DateTime;
+use Igniter\Flame\Location\Models\AbstractLocation;
+use Igniter\Local\Classes\Location;
+use Igniter\Local\Facades\Location as LocationFacade;
+use Igniter\Flame\Traits\EventEmitter;
+use Illuminate\Contracts\Events\Dispatcher;
+
+class FilterTimeslot
+{
+    use EventEmitter;
+
+    protected static $ordersCache = [];
+    protected static $scheduleCache = [];
+
+    public function subscribe(Dispatcher $dispatcher)
+    {
+        $dispatcher->listen('igniter.workingSchedule.timeslotValid', __CLASS__.'@timeslotValid');
+    }
+
+    public function timeslotValid($workingSchedule, $timeslot)
+    {
+        // Skip if the working schedule is not for delivery or pickup
+        if ($workingSchedule->getType() == AbstractLocation::OPENING)
+            return;
+            
+        $dateString = Carbon::parse($timeslot)->toDateString();
+            
+        $ordersOnThisDay = $this->getOrders($dateString);
+                
+        $location = new Location(LocationFacade::getId());
+
+        $datetime = Carbon::parse($timeslot);
+        $startTime = Carbon::parse($timeslot)->addMinutes();
```

Comment:
Did you mean to use `subMinute()` here? With `addMinutes()` the below if statement will always return false.

Reply:
I'm not sure what i was thinking to be honest, something got mixed up in the move from one approach to another. I've pushed an amended version.